internal/provider/projects_data_source.go

```diff
@@ -0,0 +1,165 @@
+								"name": schema.StringAttribute{
+									MarkdownDescription: "Name of the project",
+									Computed:            true,
+								},
+								"owned_by_workspace": schema.StringAttribute{
+									MarkdownDescription: "Workspace that owns the project",
+									Computed:            true,
+								},
+								"created_on": schema.StringAttribute{
+									MarkdownDescription: "Creation date of the project",
+									Computed:            true,
+								},
+								"deleted_on": schema.StringAttribute{
+									MarkdownDescription: "Deletion date of the project",
+									Computed:            true,
+								},
+							},
+						},
+						"spec": schema.SingleNestedAttribute{
+							MarkdownDescription: "Specifications of the project",
+							Computed:            true,
+							Attributes: map[string]schema.Attribute{
+								"display_name": schema.StringAttribute{
+									MarkdownDescription: "Display name of the project",
+									Computed:            true,
+								},
+								"tags": schema.MapAttribute{
+									MarkdownDescription: "Tags associated with the project",
+									Computed:            true,
+									ElementType:         types.ListType{ElemType: types.StringType},
+								},
+								"payment_method_identifier": schema.StringAttribute{
+									MarkdownDescription: "Payment method identifier",
+									Computed:            true,
+								},
+								"substitute_payment_method_identifier": schema.StringAttribute{
+									MarkdownDescription: "Substitute payment method identifier",
+									Computed:            true,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (d *projectsDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*MeshStackProviderClient)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *MeshStackProviderClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	d.client = client
+}
+
+func (d *projectsDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	// get workspace to query for projects
+	var workspaceIdentifier string
+	var paymentMethodIdentifier *string
+
+	resp.Diagnostics.Append(req.Config.GetAttribute(ctx, path.Root("workspace_identifier"), &workspaceIdentifier)...)
+	resp.Diagnostics.Append(req.Config.GetAttribute(ctx, path.Root("payment_method_identifier"), &paymentMethodIdentifier)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projects, err := d.client.ReadProjects(workspaceIdentifier, paymentMethodIdentifier)
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to read projects", err.Error())
+		return
+	}
+
+	var state MeshProjectsSchema
+
+	for _, p := range *projects {
+		project := MeshProject{
```

Comment:
d: `projects` is already a list of `MeshProjects`, why create new ones?

Reply:
agreed, no need for it.